### PR TITLE
Frost-coordinator/signer: Seperate public and private components of DKG rounds for speedup

### PR DESCRIPTION
**Pull request by jferrant**
Builds off of @xoloki  original code to take BIP-340 compliant SchnorrProof and use it to sign a BTC tx.
As part of @xoloki recommendations, this PR splits out the public/private calculations (the generated public key can be invalid and require re-computation, so prevent wasting unnecessary time generating known invalid private keys by first ensuring the generated public key is valid)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ members = [
 [workspace.dependencies]
 blockstack-core = { git = "https://github.com/stacks-network/stacks-blockchain/", branch = "3493-sbtc-peg-out-wire-format" }
 clap = { version = "4.1.1", features = ["derive"] }
-wtfrost = { git = "https://github.com/Trust-Machines/frost", branch = "main" }
+wtfrost = { path = "../frost" }
```
Comment (xoloki) [inline Cargo.toml:14]: Change this back to git main plz

File: frost-coordinator/src/coordinator.rs [hunk 2]
```diff
@@ -111,17 +122,33 @@ where
...
             msg: MessageTypes::NonceRequest(NonceRequest {
                 dkg_id: self.current_dkg_id,
+                sign_id: self.current_sign_id,
+                sign_nonce_id: self.current_sign_nonce_id,
             }),
             sig: [0; 32],
         };
 
-        info!("dkg_id #{}. NonceRequest sent.", self.current_dkg_id);
+        debug!("dkg_id #{}. NonceRequest sent.", self.current_dkg_id);
```
Comment (xoloki) [inline frost-coordinator/src/coordinator.rs:151]: Glad you turned down the log spam!

File: frost-coordinator/src/coordinator.rs [hunk 3]
```diff
@@ -130,57 +157,108 @@ where
...
-            let nonces: Vec<PublicNonce> = self
-                .public_nonces
-                .values()
-                .map(|n| n.nonce.clone())
-                .collect();
-            let (_, R) = compute::intermediate(msg, &ids, &nonces);
+            let R = self.compute_aggregate_nonce(msg)?;
             if R.has_even_y() {
-                info!("R has even y coord: {}", &R);
+                info!("Success: R has even y coord: {}", &R);
```
Comment (xoloki) [inline frost-coordinator/src/coordinator.rs:253]: Maybe change this to `debug!` and the failure to `warn!`?

**Review by xoloki — APPROVED** [review]
LGTM!  Fix the cargo and get tests passing, then feel free to squash/merge ))

File: Cargo.toml (hunk 1 above)
Comment (jferrant) [reply Cargo.toml:14]: Will do :)

**Comment by jferrant** [comment]
> LGTM! Fix the cargo and get tests passing, then feel free to squash/merge ))

uh oh. I think I need to update :P

**Comment by xoloki** [comment]
> > LGTM! Fix the cargo and get tests passing, then feel free to squash/merge ))
> 
> uh oh. I think I need to update :P

Test failure is almost certainly due to cargo frost path ))

File: frost-coordinator/Cargo.toml [hunk 4]
```diff
@@ -18,6 +17,7 @@ tracing = { workspace = true }
 tracing-subscriber = { workspace = true }
 frost-signer = { version = "0.0.1", path = "../frost-signer" }
 serde = { version = "1.0", features = ["serde_derive"] }
+wtfrost = { git = "https://github.com/Trust-Machines/frost", branch = "main" }
```
Comment (xoloki) [inline frost-coordinator/Cargo.toml:20]: This needs to be in the root, and use `workspace = true` here

**Review by xoloki — CHANGES_REQUESTED** [review]

**Comment by jferrant** [comment]
> > > LGTM! Fix the cargo and get tests passing, then feel free to squash/merge ))
> > 
> > 
> > uh oh. I think I need to update :P
> 
> Test failure is almost certainly due to cargo frost path ))

Oh it definitely was.

**Review by xoloki — APPROVED** [review]
LGTM, rebase then feel free to squash/merge ))

**Review by sergey-shandar — APPROVED** [review]
LGTM

**Pull request merged**